### DISCUSSION TITLE
Improve workflow to extend the structure of an existing PBPK Model (e.g. add Tumor Organ)

### DISCUSSION THREAD

**Discussion by msevestre** [Open-Systems-Pharmacology/Forum#1180] [Feature Requests & Ideas]
The current workflow to extend a PK-Sim structure in MoBi (for example in order  to add an Tumor organ to a model) could be simplified greatly. 

At the moment, the workflow consists of several steps, most of which could be automatized by MoBi:

1. Save a **representative** organ as **.pkml** file (for example Heart)
1. Import the saved organ as container and name it **Tumor** 
1. Rename all internal neighborhoods between sub-containers of imported organs
1. Modify tag of new organ & sub-containers (e.g. Rename Heart tag to Tumor)
1. Create neighborhoods for new organ (i.e, "wire" it to the blood flow)
1. Update sum parameter **Weight (tissue) in Organism**
1. Update sum parameter **Volume(plasma) in Organism**
1. Update sum parameter **Blood Flow Rate in Lung**
1. Define the partition coefficient (int/water)

I believe that steps 3, 4, 6, 7, 8, 9 should be performed automatically:

* Step 3: Rename of internal neighborhoods should happen automatically. If *Heart* was renamed to *Tumor*, then *heart_bc_heart_pls* neighborhood should also be renamed to *tumor_bc_tumor_pls* etc..
* Step 4: Renaming of Tags in subcontainers should also happen automatically
* Steps 6,7,8 are very similar. The definition of parameters **Weight (tissue) in Organism**,  **Volume(plasma) in Organism**, **Blood Flow Rate in Lung** should be changed to used a generic `SumFormula` instead of listing the list of parameters explicitly. This would have to be implemented in PK-Sim
* Step 9: The parameter is added automatically by the calculation method. Instead, this parameter should be defined all the time in the spatial structure. If it is not used, the overhead is minimal. This would have to be implemented in PK-Sim

What do you think?

**Comment by PavelBal** [comment]
This is a crucial feature for extending models. So far, the workflow is error-prone and takes much time and effort. Implementing your suggestions would help a lot.

**Comment by Yuri05** [comment]
Even more steps are required in case of protein model. All of them can be also automated via sum formulas

10.	Volume of Endosome in  EndogenousIgG/Endosome: Add endosomal tumor volume
11.	Volume of Interstitial in  EndogenousIgG/Interstitial: Add interstitial tumor volume
12.	Volume of Plasma in  EndogenousIgG/Plasma: Add plasma tumor volume
13.	Lymph flow rate in  EndogenousIgG: Add tumor lymph flow rate
14.	Fluid recirculation flow rate in EndogenousIgG: Add tumor fluid recirculation flow rate
15.	Surface area (plasma/interstitial) in the Neighborhood EndogenousIgG_int_EndogenousIgG_pls: Add respective surface area from tumor. 

Another additional steps are not related to sum formulas:

16.    Add new equation for the volume of endosomal space in tumor
        Note also that additional neighborhoods are needed with the protein model: 
        a) `Tumor endosomal space <-> EndosomalClearance` compartment
        b) `Tumor interstitial space <-> VenousBlood plasma`
17.   The property **_is present_** has to be adapted in the _Molecule Start Values_ if the tumor is added manually.

**Comment by msevestre** [comment]
@Yuri05 I really think that we should try to tackle step `9 Define the partition coefficient (int/water)` asap. This is something that is confusing for everybody and just does not make sense.

**Comment by StephanSchaller** [comment]
Great!

**Comment by wilbertdew** [comment]
Hi All,

Thanks for this excellent thread! I have been looking for a while why my tumor interstitial concentrations were different compared to the organ which was duplicated (muscle) while only the volume had changed. Turned out to be a combination of one of the points from Yuri and a new one:

16. Add new equation for the volume of endosomal space in tumor
18. Add new equation for the volume of vascular endothelium in tumor (i.e., refer to the tumor neighborhood rather then the original organ)

In addition if you change the volume of the tumor compared to the duplicated organ, the following is also relevant for the correct calculation of the isogravimetric flow if I am not mistaken:

19. Make sure the "Volume" parameter value of the tumor is identical to the "Organ volume mouse" parameter value of the tumor

If anyone has additional points/different experience or if there is a more recent thread, I would be happy to hear.

P.S. I cannot easily find this thread either via the issues or discussions search functionality, maybe possible to move this issue to the discussions?

Best,

Wilbert

**Comment by Yuri05** [comment]
>Make sure the "Volume" parameter value of the tumor is identical to the "Organ volume mouse" parameter value of the tumor

Hmm, not sure... @Christoph27 Can you comment on this?

>I cannot easily find this thread either via the issues or discussions search functionality

Show all open issues in all OSP repositories (then you can add search keywords to narrow the search results)
https://github.com/issues?q=is%3Aopen+is%3Aissue+archived%3Afalse+user%3AOpen-Systems-Pharmacology+sort%3Aupdated-desc

Show all discussions in all OSP repositories:
https://github.com/discussions?discussions_q=user%3AOpen-Systems-Pharmacology

>maybe possible to move this issue to the discussions?

Absolutely :)

**Reply by wilbertdew** [reply]
@Yuri05 I forgot to add: if you are working in the mice model. In other species you have to change it with the same factor as you changed the tumor volume compared to the duplicated organ to get the same distribution characteristics in the tumor as in the duplicated organ, I believe.  
Thanks for the links and move :)

**Reply by Yuri05** [reply]
>I forgot to add: if you are working in the mice model. In other species you have to...

This makes more sense :)

**Reply by Christoph27** [reply]
I'm not sure if I understand the discussion on point 19. together with "to get the same distribution characteristics in the tumor as in the duplicated organ". Do you want to check if the new organ was added correctly? Then I would use the same volumes (organ volume and "Organ volume mouse") as in the duplicated organ.

**Reply by wilbertdew** [reply]
Hi Christophe,

I want to make sure that in the absence of target binding, the interstitial concentrations in the tumor are the same as in the duplicated organ after I have changed the volume to a realistic tumor volume. It would not seem logical for me that the interstitial/plasma concentration ratio depends on the tumor volume in the absence of target binding. If you only change the organ volume, and not the "organ volume mouse" parameter compared to the duplicated tissue, then the isometric flow is calculated as it is needed for different species according to the Vorg/Vorg,mouse ratio and the scaling exponent, it seems.
